src/django_components/slots.py

```diff
@@ -227,75 +395,181 @@ def __init__(
         kwargs: RuntimeKwargs,
         trace_id: str,
         node_id: Optional[str] = None,
-        is_implicit: bool = False,
     ):
         super().__init__(nodelist=nodelist, args=None, kwargs=kwargs, node_id=node_id)
 
-        self.is_implicit = is_implicit
         self.trace_id = trace_id
-        self.component_id: Optional[str] = None
 
     def render(self, context: Context) -> str:
+        if self._is_extracting_fill(context):
```

Comment:
Similarly to slots, also fills are resolve at render-time.

1. We render the content between `{% component %} .. {% endcomponent %}` as `NodeList.render()`.
2. We also set a special variable `FILL_GEN_CONTEXT_KEY` to the context
3. When we come across a `{% fill %}` tag, we enter `FillNode.render()`. Here, we check if we're in the "fill collection mode" by checking for the special variable.
4. If so, we extract the data like the fill name, and push it to a list of fills that's available under `FILL_GEN_CONTEXT_KEY`.
5. Once we're done rendering, the list that was set under `FILL_GEN_CONTEXT_KEY` is now populated with the fills that we've discovered during rendering.